mqtt/mqtt-bridge/src/persist/waking_state/waking_store.rs

```diff
@@ -0,0 +1,97 @@
+use std::collections::hash_map::Entry;
+use std::{boxed::Box, collections::HashMap, task::Waker};
+
+use bincode::{deserialize, serialize, ErrorKind};
+use mqtt3::proto::Publication;
+use rocksdb::{IteratorMode, DB};
+use tracing::error;
+
+use crate::persist::{waking_state::StreamWakeableState, Key, PersistError};
+
+/// Responsible for waking waiting streams when new elements are added.
+///
+/// Exposes a get method for retrieving a count of elements in order of insertion.
+/// When elements are retrieved they are added to the in flight collection, but kept in the original db store.
+/// Only when elements are removed from the in-flight collection they will be removed from the store.
+pub struct WakingStore {
+    db: DB,
```

Comment:
initial idea was we have only on DB per all bridges. For each bridge there will be 2 collections of publication in and out. with this approach you have right now for each bridge there will be a separate DB which seems like overkill.

Reply:
I agree that we should share the db. So we need to wrap in a mutex?

Reply:
Since DB is already Send / Sync I will use an Arc.

Reply:
Although now that we are reusing the same DB struct, there are other things I need to change. For example:
- how do we handle offset
- how do we only get/insert elements for the store in question

Reply:
We can have a column family for each store based on a generated uuid. Offset will be specific to each store as described in the spec.